### DISCUSSION TITLE
Visual Particle Editor for Godot 3.6/Godot 4?

### DISCUSSION THREAD

**Discussion by MangoDragonHub** [godotengine/godot-proposals#4712] [Engine Core]
I originally was thinking it would be cool to get an overhauled version of the current Visual Scripting system Godot already, but I remembered asking this question a long time ago and was told there wasn't anyone make addition changes to it. So I thought of a better compromise, a Visual Particle Editor for both GPU and CPU particles.

The way how I'm looking at it, is similar to the current Visual Scripting system for the Material Editor where you create dynamical and modular parameters for each effect that you want. Sometimes, the standard functionally of the current particle system editor might not be enough to create effects that people might want to make, but with this it can give people the option having a more flexible and dynamic workflow to creating VFXs.

**Comment by Calinou** [comment]
There's already a visual shader editor, which can be used to create custom particle shaders: https://github.com/godotengine/godot/pull/41840, https://github.com/godotengine/godot/pull/42248

CPUParticles is much more limited in terms of functionality, and it's not possible to create custom behavior with it.
